src/storyMap/components/StoryMap.js

```diff
@@ -0,0 +1,543 @@
+      const layerType = map.getLayer(layer).type;
+      return LAYER_TYPES[layerType];
+    },
+    [map]
+  );
+
+  const setLayerOpacity = useCallback(
+    layer => {
+      const paintProps = getLayerPaintType(layer.layer);
+      paintProps.forEach(function (prop) {
+        const options = layer.duration ? { duration: layer.duration } : {};
+        if (layer.duration) {
+          const transitionProp = prop + '-transition';
+          map.setPaintProperty(layer.layer, transitionProp, options);
+        }
+        map.setPaintProperty(layer.layer, prop, layer.opacity, options);
+      });
+    },
+    [map, getLayerPaintType]
+  );
+
+  const initialLocation = useMemo(() => {
+    if (config.titleTransition?.location) {
+      return config.titleTransition?.location;
+    }
+    const firstChapterWithLocation = config.chapters.find(
+      chapter => chapter.location
+    );
+    return firstChapterWithLocation?.location;
+  }, [config.chapters, config.titleTransition?.location]);
+
+  useEffect(() => {
+    const map = new mapboxgl.Map({
+      container: mapContainer.current,
+      style: config.style,
+      interactive: false,
+      transformRequest: transformRequest,
+      projection: config.projection || 'globe',
+      zoom: 1,
+      ...(initialLocation ? initialLocation : {}),
+    });
+
+    map.on('load', function () {
+      if (config.use3dTerrain) {
+        map.addSource('mapbox-dem', {
+          type: 'raster-dem',
+          url: 'mapbox://mapbox.mapbox-terrain-dem-v1',
+          tileSize: 512,
+          maxzoom: 14,
+        });
+        // add the DEM source as a terrain layer with exaggerated height
+        map.setTerrain({ source: 'mapbox-dem', exaggeration: 1.5 });
+
+        // add a sky layer that will show when the map is highly pitched
+        map.addLayer({
+          id: 'sky',
+          type: 'sky',
+          paint: {
+            'sky-type': 'atmosphere',
+            'sky-atmosphere-sun': [0.0, 0.0],
+            'sky-atmosphere-sun-intensity': 15,
+          },
+        });
+      }
+
+      setMap(map);
+    });
+    map.on('style.load', () => {
+      map.setFog({
+        color: 'rgb(169, 169, 188)', // Lower atmosphere
+        'high-color': 'rgb(16, 16, 20)', // Upper atmosphere
+        'horizon-blend': 0.02, // Atmosphere thickness (default 0.2 at low zooms)
+        'space-color': 'rgb(20, 20, 26)', // Background color
+        'star-intensity': 0.1, // Background star brightness (default 0.35 at low zoooms )
+      });
+    });
+    return () => map.remove();
+  }, [config.style, initialLocation, config.use3dTerrain, config.projection]);
+
+  useEffect(() => {
+    if (!map || !config.inset) return;
+
+    const newInsetMap = new mapboxgl.Map({
+      container: mapInsetContainer.current,
+      style: 'mapbox://styles/mapbox/dark-v10', //hosted style id
+      center: config.chapters[0].location.center,
+      // Hardcode above center value if you want insetMap to be static.
```

Comment:
@paulschreiber That came from the original code, see: https://github.com/mapbox/storytelling/blob/main/src/index.html#L289 What it means is that if using the inset map you don't set it to use it form the chapter it sill be static showing the center position you define there. Since this code was translated I didn't wnat to lose the comments from the original code, because we will have to maintain it in the future

Reply:
Please expand the comment to include this information.